sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/EventHubSharedKeyCredential.cs

```diff
@@ -72,6 +78,28 @@ public sealed class EventHubSharedKeyCredential : TokenCredential
         ///
         public override ValueTask<AccessToken> GetTokenAsync(TokenRequestContext requestContext, CancellationToken cancellationToken) => throw new InvalidOperationException(Resources.SharedKeyCredentialCannotGenerateTokens);
 
+        /// <summary>
+        ///   Allows the rotation of Shared Access Signatures.
+        /// </summary>
+        ///
+        /// <param name="keyName">The Event Hubs resource to which the token is intended to serve as authorization.</param>
+        /// <param name="keyValue">The duration that the signature should be considered valid; if not specified, a default will be assumed.</param>
+        /// <param name="signatureValidityDuration">The duration that the signature should be considered valid; if not specified, a default will be assumed.</param>
```

Comment:
We're not passing `signatureValidityDuration` along.   We should be sending that to generate the key, right?

Reply:
Good point. 

My intention was to not pass ```signatureValidityDuration```and just reuse the existing expiration.

I was considering that for Shared Access Credentials the token is automatically refreshed so ```signatureValidityDuration``` might be a detail that would not add much the caller?

But definitely I might be missing some important detail?

Reply:
I think we're good with not passing it if we're reverting behavior to generating on demand as we were discussing.   This was more poking at the "we're making it and potentially caching with a stale date."